### PR TITLE
Clarify custom link color documentation for themes

### PR DESCRIPTION
**Pull request by kjellr**
[Following up on](https://github.com/WordPress/gutenberg/pull/23590#issuecomment-658804519) #23590, this PR clarifies how themes can properly opt-in and support the experimental custom link color. 

@nosolosw, does this look about right to you?

### CODE REVIEW

**Review by oandregal — APPROVED** [review]
This sounds right, thanks!

**Comment by metalandcoffee** [comment]
@kjellr Hi there! I'm actually testing this feature right now and I'm finding the linked documentation a bit hard to follow. It might be a good idea to provide a `experimental-theme.json` example (with defined link colors) for good measure especially since there isn't really an example in the linked documentation.

> The theme should also ensure that its link styles map to the `--wp--style--color--link` variable in both the editor and front-end stylesheets. 

A more detailed explanation here would be helpful as well since I find this confusing. Is this a step you have to do if you do not have an `experimental-theme.json` file?

**Comment by nosolosw** [comment]
Would this update make this clearer?

---

If a theme opts in, it should [define default link colors](https://developer.wordpress.org/block-editor/developers/themes/theme-json/#color-properties) in `experimental-theme.json` (or in its theme styles if no `experimental-theme.json` is present). For example:

```css
{ 
    "global": {
        "styles": {
            "color": {
                "link": "hotpink"
            }
        }
    }
}
```

The theme should also ensure that its link styles map to the `--wp--style--color--link` variable in both the editor and front-end stylesheets:

```css
a {
    color: var(--wp--style--color--link);
}
```

**Comment by metalandcoffee** [comment]
@nosolosw That's perfect! Thank you for that!

**Comment by kjellr** [comment]
Excellent suggestion, thanks @metalandcoffee and @nosolosw! I've added that in f988afc and will merge in as soon as the tests pass.
